src/main.rs

```diff
@@ -172,6 +194,36 @@ impl RelayTask {
         Ok(())
     }
 }
+
+async fn handle_host_ip_provider_conn(
+    mut stream: EnclaveStream,
+    host_ip_address: &str,
+) -> Result<()> {
+    stream.write_all(host_ip_address.as_bytes()).await?;
+    stream.flush().await?;
+    Ok(())
+}
+
+async fn host_ip_provider_server(port: u16) -> Result<()> {
+    let host_ip_address = local_ip_address::local_ip()?.to_string();
```

Comment:
> Is there a design doc for this?

I plan to update the [key sync doc](https://github.com/brave/nitriding-daemon/blob/master/doc/key-synchronization.md) in a future nitriding-daemon PR.

I'm looking to replace this function: https://github.com/brave/nitriding-daemon/blob/445872d1b5ca76d327a0703399cb4330ec86f981/util.go#L133-L137

The IP returned from this function is sent to the leader enclave, so the leader know where's to contact the worker.

> local_ip relies on a somewhat heuristic approach of finding the address of an interface which routes to the "reserved for documentation" 192.0.2.0 on linux. That seems likely to find an external network, but is also brittle. Hosts have multiple interfaces, and if this is for inter-enclave key exchange don't we want to select the private interface rather than the public one?

A simple enclave host (which include containerized hosts in k8s) will have three interfaces: the loopback interface, the VPC-facing interface and the vsock interface. Therefore, it should be pretty straightforward for the local IP crate to find the correct IP that is in the VPC subnet. There is no 'public' interface, as the NLB (which faces the internet of course) will forward requests to the private, VPC-facing interface.

>Seems like it would be better to just take the host address as an argument and have the deployment script configure it directly?

This would be a little difficult to do, since the pod IPs are dynamic and not managed by us.